plus/src/main/java/org/apache/calcite/adapter/os/SqlShell.java

```diff
@@ -64,6 +64,10 @@ public class SqlShell {
     this.err = Objects.requireNonNull(err, "err");
   }
 
+  public static void setModel(String newModel) {
```

Comment:
Does this really need to be mutable?

Could you achieve what you need with a `withModel(String model)` method?